packages/flutter/lib/src/gestures/long_press.dart

```diff
@@ -291,34 +350,55 @@ class LongPressGestureRecognizer extends PrimaryPointerGestureRecognizer {
   }
 
   void _checkLongPressStart() {
-    assert(_initialButtons == kPrimaryButton);
-    if (onLongPressStart != null) {
-      final LongPressStartDetails details = LongPressStartDetails(
-        globalPosition: _longPressOrigin.global,
-        localPosition: _longPressOrigin.local,
-      );
-      invokeCallback<void>('onLongPressStart',
-        () => onLongPressStart(details));
+    assert(_initialButtons == kPrimaryButton || _initialButtons == kSecondaryButton);
```

Comment:
Can we move it to a `switch` and assert(false) at default? (Same for other handlers)

Reply:
Done.